pkgs/top-level/all-packages.nix

```diff
@@ -12076,7 +12076,11 @@ with pkgs;
     withUkify = true;
   };
 
-  udev = if lib.meta.availableOn stdenv.hostPlatform systemdLibs then systemdLibs else libudev-zero;
+  udev =
+    if (lib.meta.availableOn stdenv.hostPlatform systemdLibs) && (config.withSystemd or true) then
```

Comment:
@winterqt `config.systemd` is defined here. It's not necessary to add it to the `config` attrset and doing so would increase the scope of this PR and wedge it.

Reply:
@ehmry I don’t know what you mean. Since when do we condition on arbitrary keys in `config` that aren’t documented or defined within the main attrset with a default and a description?

Reply:
Agreed. We can't just add a sneaky, undocumented `config` option here. Additionally, it can't be so weakly implemented. If we're going to have a `config.withSystemd`, then it should *ensure* that systemd is not used. i.e. It should be a change to `systemd/default.nix` that results in `lib.meta.availableOn` returning false no matter what when `config.withSystemd = false`. That way we don't give users a false sense of having disabled systemd when in fact it will easily creep into their closures. Instead they'll get justified eval errors. And of course by making it `availableOn == false`, that means we don't need to change this condition.